test/example.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { exec } from 'child_process';
+import { describe, expect, test } from 'vitest';
+
+async function generateExample(name, args = '') {
+  const fixture = `./test/fixture/${name}.yaml`;
+  const output = `./example/test-${name}${args.replace(/\s+/g, '-')}.js`;
+  const command = `node ./dist/cli.js ${fixture} --output ${output} ${args}`;
+  await exec(command, { cwd: '.' });
+  return output;
+}
+
+describe('example', () => {
+  test.each([
+    { name: 'githubapi', args: '--node' },
+    { name: 'githubapi', args: '--react-native' },
+  ])('$name $args should be valid', async ({ name, args }) => {
+    const example = await generateExample(name, args);
+    await expect(import(example)).resolves.toMatchObject({});
```

Comment:
can this switch to `toMatchSnapshot`? it will generate a snapshot file for comparison next time, it's weird to match an empty object here.

https://vitest.dev/guide/snapshot.html

Reply:
Initially, the test only checked if the validity of the generated code, last commit new ensures the output does not change.

Maybe a note could be added with [instructions to update snapshots](https://vitest.dev/guide/snapshot.html#updating-snapshots) and remind contributors to use `pnpm`.

Reply:
The snapshot contains local absolute paths, maybe `info.callFrame` can be made relative.

Side note `exec` is not working as I expected, I may revert to `execSync`.

Reply:
Snapshots should work on any system with the last commit, making the paths relative to `process.cwd()` and a bit of `\\` gymnastics for Windows.

The examples are generated before each test, maybe they could be cached if nothing changed. I had to increase the timeout because the generation time is now included in the test.